apps/examples/board_specific_demo/ameba_wifi_csi_demo/example_wifi_csi.c

```diff
@@ -0,0 +1,142 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+
+  ******************************************************************************/
+
+
+#include <stdio.h>
+#include <tinyara/wifi/rtk/wifi_structures.h>
+#include "example_wifi_csi.h"
+#include "sched.h"
+
+#define IP_ADDR_INVALID "0.0.0.0"
+static uint8_t * getIp(void);
+extern int wifi_csi_config(rtw_csi_action_parm_t *act_param);
+extern int wifi_csi_report(uint32_t buf_len, uint8_t *csi_buf, uint32_t *len, rtw_csi_header_t *csi_header);
+extern int wifi_is_running(unsigned char wlan_idx);
+sem_t wc_ready_sema;
+
+unsigned int csi_raw_data_len = 2048;
+int _wifi_csi_thread(int argc, char *argv[])
+{
+	rtw_csi_action_parm_t act_param = {0};
+	rtw_csi_header_t csi_hdr = {0};
+	unsigned int len;
+	unsigned char *csi_buf = NULL;
+	act_param.group_num = 0;
+	act_param.mode = 2;  /* currently only supported */
+	act_param.accuracy = 0;
+	act_param.trig_period = 200;  /* ms */
+	act_param.data_rate = 0xC;  /* ofdm 6 mpbs*/
+//	act_param.mac_addr = {0x00, 0xe0, 0x4c, 0x81, 0x92, 0xbb};
+
+	while (1) {
+		uint8_t *ip = getIp();
+		if (wifi_is_running(WLAN0_IDX) && ((wifi_get_join_status() == RTW_JOINSTATUS_SUCCESS) && (strncmp(ip, IP_ADDR_INVALID, sizeof(ip))))){
+			up_mdelay((unsigned long)2000); /* 2s */
+			break;
+		}
+		up_mdelay((unsigned long)2000); /* 2s */
+	}
+
+	/* register wifi event callback function */
+	wifi_reg_event_handler(WIFI_EVENT_CSI_DONE, example_wifi_csi_report_cb, NULL);
+
+	/**
+	 * should use semaphore to wait wifi event happen
+	 * the following example shows that we wait for wifi csi ready
+	 */
+	rtw_init_sema(&wc_ready_sema, 0);
+
+	/* csi cfg and csi en */
+	act_param.act = 1;  /* csi cfg */
+	act_param.enable = 0;
+	wifi_csi_config(&act_param);
+
+	act_param.act = 0;  /* csi en */
+	act_param.enable = 1;
+	wifi_csi_config(&act_param);
+
+	while (1) {
+		/* example: when wifi csi rx done, call csi report handle function. */
+		if (rtw_down_sema(&wc_ready_sema) == RTW_FALSE) {
+			rtw_free_sema(&wc_ready_sema);
+			act_param.act = 0;  /* csi dis */
+			act_param.enable = 0;
+			wifi_csi_config(&act_param);
+			break;
+		}
+
+		csi_buf = rtw_malloc(csi_raw_data_len);
+
+		if (csi_buf != NULL) {
+			wifi_csi_report(csi_raw_data_len, csi_buf, &len, &csi_hdr);
+
+			/*do something for handing csi info*/
+			printf("\n CSI data extracted from the peer package:\n");
+			for (int i = 0; i < len; i++){
+				printf("%02X ", *(csi_buf + i ));
+			}
+				printf("\n");
+
+		} else {
+			printf("\n csi_buf malloc fail\n");
+		}
+
+		if (csi_buf != NULL) {
+			rtw_mfree(csi_buf, csi_raw_data_len);
+		}
+	}
+
+	/* unregister wifi event callback function */
+	wifi_unreg_event_handler(WIFI_EVENT_CSI_DONE, example_wifi_csi_report_cb);
+
+	rtw_free_sema(&wc_ready_sema);
+
+
+	rtw_delete_task(NULL);

```

Comment:
Unnecessary. Let's remove.

Reply:
Correct. The deleting task at end of the function is unnecessary. It's to terminating the task at middle of operation. But yours already finishes all of operation and will be deleted automatically.